signalduino_protocols.hash - minify

HomeAutoUser opened the pull request:
- added minify
https://github.com/RFD-FHEM/RFFHEM/issues/114

* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

HomeAutoUser commented:
@sidey79 
beim durchschauen ist mir aufgefallen, das wir die Vorlage für einen Issues noch ergänzen können mit den Hinweisen
1) exakte Hersteller - Typen - Modellbezeichnungen
2) wenn möglich, bitte Bilder des Gerätes oder / und dessen Platine

sidey79 commented:
@HomeAutoUser 

Können wir machen ja. 

Was das Protokoll 19 angeht, wäre da jetzt ein developid  =y von nöten?

HomeAutoUser commented:
>Was das Protokoll 19 angeht, wäre da jetzt ein developid =y von nöten?

Das neue Protokoll habe ich nicht mit einem 
>developid =y

versehen weil uns da die Nachrichten zum testen fehlen. Die remote hat so viele Tasten und da bräuchten wir mehr Informationen. Aus diesem Grund habe ich es als u19 gelassen um auf jedenfall zu gewährleisten wenn jemanch dies hat, sofort mit einem Notify auszuwerten.

Sollten wir an mehr Daten kommen, so würde ich dann ein y setzen wollen wenn wir ggf. es weiter einbauen.

sidey79 commented:
ok verstehe.  Das hilft uns so natürlich mehr.

HomeAutoUser commented:
Wo soll der Fehler auftreten?
Ich habe es mit beiden Nachrichten dispatcht ohne Error.

HomeAutoUser commented:
>not ok 17 - check not called Dispatch from parse MU

Kann es sein, das du damit sagen möchtest, es gibt keine Weitergabe?
Ein modul gibt es auf jedenfall nicht.

sidey79 commented:
Hmm, der Test erwartet dass es auf die Nachricht

`MU;P0=-563;P1=479;P2=991;P3=-423;P4=361;P5=-1053;P6=3008;P7=-7110;D=2345454523452323454523452323452323452323454545456720151515201520201515201520201520201520201515151567201515152015202015152015202015202015202015151515672015151520152020151520152020152020152020151515156720151515201520201515201520201520201520201515151;CP=1;R=21;`

Keinen Dispatch gibt. Jetzt gibt es aber einen. Alledings bei Protokoll 5 und 9 .. Irgendwie kommt mir der Test selber jetzt auch seltsam vor.

HomeAutoUser commented:
Das klärt es vermutlich auf bzw. wieso es zum Fehler kam vielleicht.

```
2018.10.04 22:50:18 5: sduino_dummy/msg adding start and endmarker to message
2018.10.04 22:50:18 4: sduino_dummy/msg get raw: MU;P0=-563;P1=479;P2=991;P3=-423;P4=361;P5=-1053;P6=3008;P7=-7110;D=2345454523452323454523452323452323452323454545456720151515201520201515201520201520201520201515151567201515152015202015152015202015202015202015151515672015151520152020151520152020152020152020151515156720151515201520201515201520201520201520201515151;CP=1;R=21;
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 5 -> unitec6899 matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: Starting demodulation at Position 50
2018.10.04 22:50:18 5: sduino_dummy: dispatching bits: 1 0 0 0 1 0 1 1 0 0 1 0 1 1 0 1 1 0 1 1 0 0 0 0
2018.10.04 22:50:18 4: sduino_dummy: decoded matched MU Protocol id 5 dmsg i8B2DB0 length 24 RSSI = -63.5
2018.10.04 22:50:18 5: sduino_dummy Dispatch: i8B2DB0, test ungleich: disabled
2018.10.04 22:50:18 5: sduino_dummy Dispatch: i8B2DB0, -63.5 dB, dispatch
2018.10.04 22:50:18 5: sduino_dummy: dispatch i8B2DB0
2018.10.04 22:50:18 4: sduino_dummy IT: message "i8B2DB0" (7)
2018.10.04 22:50:18 4: sduino_dummy IT: msgcode "" (0) bin = 100010110010110110110000
2018.10.04 22:50:18 5: sduino_dummy IT: EV1527 housecode = 1527x8b2db  onoffcode = 0000
2018.10.04 22:50:18 4: sduino_dummy IT: 1527x8b2db not defined (Switch code: 0000)
2018.10.04 22:50:18 5: sduino_dummy: dispatching bits: 1 0 0 0 1 0 1 1 0 0 1 0 1 1 0 1 1 0 1 1 0 0 0 0
2018.10.04 22:50:18 4: sduino_dummy: decoded matched MU Protocol id 5 dmsg i8B2DB0 length 24 RSSI = -63.5
2018.10.04 22:50:18 5: sduino_dummy Dispatch: i8B2DB0, test gleich
2018.10.04 22:50:18 4: sduino_dummy Dispatch: i8B2DB0, Dropped due to short time or equal msg
2018.10.04 22:50:18 5: sduino_dummy: dispatching bits: 1 0 0 0 1 0 1 1 0 0 1 0 1 1 0 1 1 0 1 1 0 0 0 0
2018.10.04 22:50:18 4: sduino_dummy: decoded matched MU Protocol id 5 dmsg i8B2DB0 length 24 RSSI = -63.5
2018.10.04 22:50:18 5: sduino_dummy Dispatch: i8B2DB0, test gleich
2018.10.04 22:50:18 4: sduino_dummy Dispatch: i8B2DB0, Dropped due to short time or equal msg
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 8 -> TX3 Protocol matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: Starting demodulation at Position 50
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 9 -> CTW 600 matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: Starting demodulation at Position 50
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 19 -> minify matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: Starting demodulation at Position 0
2018.10.04 22:50:18 5: sduino_dummy: dispatching bits: 1 0 0 0 1 0 1 1 0 0 1 0 1 1 0 1 1 0 1 1 0 0 0 0
2018.10.04 22:50:18 4: sduino_dummy: decoded matched MU Protocol id 19 dmsg u19#8B2DB0 length 24 RSSI = -63.5
2018.10.04 22:50:18 5: sduino_dummy Dispatch: u19#8B2DB0, test ungleich: disabled
2018.10.04 22:50:18 5: sduino_dummy Dispatch: u19#8B2DB0, -63.5 dB, dispatch
2018.10.04 22:50:18 5: sduino_dummy: dispatch u19#8B2DB0
2018.10.04 22:50:18 4: SIGNALduino_unknown incomming msg: u19#8B2DB0
2018.10.04 22:50:18 4: SIGNALduino_unknown rawData: 8B2DB0
2018.10.04 22:50:18 4: SIGNALduino_unknown Protocol: 19
2018.10.04 22:50:18 4: SIGNALduino_unknown converted to bits: 100010110010110110110000
2018.10.04 22:50:18 4: SIGNALduino_unknown sduino_dummy Protocol:19 | To help decode or debug, please add u19! (attr sduino_dummy development u0,m30,y73,y73.1,y74,m81,m83,u85,m72,u19)
2018.10.04 22:50:18 4: Unknown, please report
2018.10.04 22:50:18 4: SIGNALduino_unknown incomming msg: u19#8B2DB0
2018.10.04 22:50:18 4: SIGNALduino_unknown rawData: 8B2DB0
2018.10.04 22:50:18 4: SIGNALduino_unknown Protocol: 19
2018.10.04 22:50:18 4: SIGNALduino_unknown converted to bits: 100010110010110110110000
2018.10.04 22:50:18 4: SIGNALduino_unknown sduino_dummy Protocol:19 | To help decode or debug, please add u19! (attr sduino_dummy development u0,m30,y73,y73.1,y74,m81,m83,u85,m72,u19)
2018.10.04 22:50:18 4: Unknown, please report
2018.10.04 22:50:18 3: sduino_dummy: Unknown code u19#8B2DB0, help me!
2018.10.04 22:50:18 5: sduino_dummy: applying filterfunc SIGNALduino_filterSign
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 21 -> einhell garagedoor matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: start pattern for MU Protocol id 21 -> einhell garagedoor mismatches, aborting
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 26 -> remote26 matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: start pattern for MU Protocol id 26 -> remote26 mismatches, aborting
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 27 -> remote27 matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: Starting demodulation at Position 2
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 28 -> IC Ledspot matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: start pattern for MU Protocol id 28 -> IC Ledspot mismatches, aborting
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 29 -> HT12e remote matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: start pattern for MU Protocol id 29 -> HT12e remote mismatches, aborting
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 30 -> diverse matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: start pattern for MU Protocol id 30 -> diverse mismatches, aborting
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 31 -> pollin isotronic matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: Starting demodulation at Position 53
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 36 -> socket36 matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: start pattern for MU Protocol id 36 -> socket36 mismatches, aborting
2018.10.04 22:50:18 5: sduino_dummy: applying filterfunc SIGNALduino_compPattern
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 49 -> quigg_gt9000 matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: Starting demodulation at Position 2
2018.10.04 22:50:18 5: sduino_dummy: dispatching bits: 0 0 0 1 0 1 1 0 0 1 0 1 1 0 1 1 0 1 1 0 0 0 0 0
2018.10.04 22:50:18 4: sduino_dummy: decoded matched MU Protocol id 49 dmsg U49#165B60 length 24 RSSI = -63.5
2018.10.04 22:50:18 5: sduino_dummy Dispatch: U49#165B60, test ungleich: disabled
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 50 -> optus_XT300 matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: Starting demodulation at Position 50
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 59 -> AK-HD-4 matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: start pattern for MU Protocol id 59 -> AK-HD-4 mismatches, aborting
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 60 -> WS2000 matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: Starting demodulation at Position 0
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 64 -> WH2 matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: Starting demodulation at Position 50
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 69 -> Hoermann matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: start pattern for MU Protocol id 69 -> Hoermann mismatches, aborting
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 72 -> Siro shutter matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: start pattern for MU Protocol id 72 -> Siro shutter mismatches, aborting
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 75 -> ConradRSL2 matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: Starting demodulation at Position 50
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 79 -> VTX-BELL_Funkklingel matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: start pattern for MU Protocol id 79 -> VTX-BELL_Funkklingel mismatches, aborting
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 81 -> SA-434-1 matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: start pattern for MU Protocol id 81 -> SA-434-1 mismatches, aborting
2018.10.04 22:50:18 4: sduino_dummy: Fingerprint for MU Protocol id 83 -> RH787T matches, trying to demodulate
2018.10.04 22:50:18 5: sduino_dummy: start pattern for MU Protocol id 83 -> RH787T mismatches, aborting
```

HomeAutoUser commented:
@sidey79 
bestimmt weil er ID5 dispatchen möchte.

https://github.com/RFD-FHEM/RFFHEM/issues/308 Bei uns ist es kein I sondern p.
In dem o.g. Issues habe ich es bemerkt, das dort Differenzen sind.

... aber da muss ich nochmal auf die Definition schauen wieso diese auf einmal mitkommt.
Hier haben wir wieder angeblich "so viel unterschied" aber der interprätiert es mit als ein ähnliches protokoll.

```
    "5"    => 	## Similar protocol as intertechno, but without sync
				# ! same definition how ID 75, but other length !
				# MU;P0=-404;P1=1319;P2=-1022;P3=546;P4=-7696;D=012121212321232123232321232323212121212121232323212321212121212121232323212321212323212140;CP=3;R=230;
        {
            name			=> 'unitec6899',	
			id          	=> '5',
			one				=> [3,-1],
			zero			=> [1,-3],
			clockabs     	=> 500,			# -1 = auto
			format 			=> 'twostate',	# tristate can't be migrated from bin into hex!
			preamble		=> 'i',			# Append to converted message
			clientmodule    => 'IT',   		# not used now
			modulematch     => '^i......',  # not used now
			length_min      => '24',

		},  
```


```
	"19" => # minify Funksteckdose
            # https://github.com/RFD-FHEM/RFFHEM/issues/114
			# MU;P0=293;P1=-887;P2=-312;P6=-1900;P7=872;D=6727272010101720172720101720172010172727272720;CP=0;
			# MU;P0=9078;P1=-308;P2=180;P3=-835;P4=881;P5=309;P6=-1316;D=0123414141535353415341415353415341535341414141415603;CP=5;
		{
            name			=> 'minify',
			comment 		=> 'RC202 remote ?',
			id          	=> '19',
			one				=> [3,-1],
			zero			=> [1,-3],
			clockabs		=> 300,
			format 			=> 'twostate',	  		
			preamble		=> 'u19#',				# prepend to converted message
			#clientmodule    => '',   				# not used now
			#modulematch     => '',  				# not used now
			length_min      => '19',
			#length_max      => '32',
		}, 	 	
```

HomeAutoUser commented:
Komisch, ich habe nun zum erneuten mal, beide Nachrichten Dispatcht und diesmal erhielt ich KEIN Ansprechen auf ID5. Vermutlich ist das genau die Grenze...

sidey79 commented:
Die Toleranz verändert sich aber nicht.

sidey79 commented:
Ahh langsam blicke ich durch.

Das Protokoll 5 ist eigentlich kaputt:

Irgendwann war modulematch mal ohne Funktion. Da konnte das Protokoll 5 auch dispatcht werden.
```
			preamble		=> 'p5#',			# Append to converted message	
			clientmodule    => 'IT',   		# not used now
			modulematch     => '^i......',  # not used now
```

Diese Logzeile hat mich jetzt in die irre geführt. Da steht dass etwas dekodiert wurde.
`2018.10.04 19:28:38 4: dummyDuino: decoded matched MU Protocol id 5 dmsg p5#8B2DB0 length 24 RSSI = -63.5`

Da steht nicht, dass es dispatcht wurde. Geht auch nicht, da modulematch nicht stimmt. Leider gibt es dafür keine Logausgabe


Aus deinem Log werde ich nicht schlau. Du musst den Protocol Hash für ID5 angepasst haben. Sonst kann es nicht zu diesem Effekt kommen.
`decoded matched MU Protocol id 5 dmsg i8B2DB0 length 24 RSSI = -63.5`

HomeAutoUser commented:
Aus deinem Log werde ich nicht schlau. Du musst den Protocol Hash für ID5 angepasst haben. Sonst kann es nicht zu diesem Effekt kommen.
>decoded matched MU Protocol id 5 dmsg i8B2DB0 length 24 RSSI = -63.5

Das ist richtig weil ich es getestet habe mit beiden Varianten. Ich habe das Log mit der korrigierten Version erstellt.

Um durchzusehen zwecks Protokoll 5 was eigentlich kaputt ist nocht, schaue die Gegenüberstellung https://github.com/RFD-FHEM/RFFHEM/issues/308 an. Da siehst du den Ist-Stand und auch die Korrektur welche schon Ralf drin hat. Wir müssen dies als ebenso korrigieren.

Es gibt aber auch noch 2 andere Änderungen bei 2 anderen Protokollen.

sidey79 commented:
@homeautouser

Gut. Dann ist hier ja alles in Ordnung.
Für Protokoll #5 können wir dann ja auch einen Test erstellen, da wir ja das Ergebnis der Parse Funktion kennen :)
